custodia/httpd/server.py

```diff
@@ -436,25 +464,39 @@ class HTTPServer(object):
 
     def __init__(self, srvurl, config):
         url = urlparse(srvurl)
-        address = unquote(url.netloc)
         if url.scheme == 'http+unix':
             # Unix socket
+            address = None
+            if sd_listen_fds is not None:
```

Comment:
SSSD can't even use Unix Socket in the proxy provider yet ...
The comment was that we should get the code in shape so it can properly do that check.

With the current code you could configure systemd to listen on a TCP socket, and custodia.conf to say http+unix and you would end up serving via a TCP socket while thinking it is a unix socket anyway.

Of course things like getpeercreds wil fail later, if it is used at all.

Anyway getsockopt() can be used the same way we do here:
https://github.com/latchset/custodia/blob/master/custodia/httpd/server.py#L199
for getpeercreds.

We should do that, in this PR, until we have an officially supported pythonic way to do it.